packages/client-app/src/views/Servers/ServerForm.vue

```diff
@@ -33,12 +33,55 @@ const updateServer = (key: string, value: string) => {
   if (!activeCollection.value) return
   serverMutators.edit(activeServer.value.uid, key as keyof Server, value)
 }
+
+const updateVariable = (key: string, value: any) => {
+  if (!activeCollection.value) return
+  serverMutators.edit(activeServer.value.uid, `variables.${key}.value`, value)
+}
+
+const variableOptions = computed(() => {
+  const variables = activeServer.value?.variables ?? {}
+
+  const keys = Object.keys(variables)
+
+  return keys.map((key: string) => ({
+    key,
+    label: key,
+    placeholder: variables?.[key]?.default ?? '',
+  }))
+})
```

Comment:
Object.entries would tidy this up a bit.

```suggestion
const variableOptions = computed(() =>
  Object.entries(activeServer.value?.variables ?? {}).map(([key, val]) => ({
    key,
    label: key,
    placeholder: val ?? '',
  })),
)
```